### ISSUE TITLE
Rate Limit errors are now HTML instead of JSON

### ISSUE THREAD

**Issue by nozzlegear** [nozzlegear/ShopifySharp#236]
Shopify has made an **unannounced change** where requests that go over the rate limit now return an HTML error page instead of the JSON they used to return. Adding an "accept: application/json" request header doesn't change the response.

**Comment by nozzlegear** [comment]
I'm now seeing HTML returned for most OAuth errors, such as when trying to reuse an authorization code.

**Comment by clement911** [comment]
Just to clarify, the only impact is the error message being returned?

**Comment by nozzlegear** [comment]
Yes I think so, the package is still throwing a ShopifyException with the correct status code. The only problem is instead of real error messages on the exception's .Errors prop, it's instead filled with the message thrown by Json.Net trying to deserialize HTML. So we lose helpful error messages and instead get something unrelated.




--


Josh Harms







From: Clement Gutel


Sent: Tuesday, April 17, 18:26


Subject: Re: [nozzlegear/ShopifySharp] Rate Limit errors are now HTML instead of JSON (#236)


To: nozzlegear/ShopifySharp


Cc: Joshua Harms, State change






Just to clarify, the only impact is the error message being returned?

**Comment by clement911** [comment]
Hopefully it shouldn't break any applications, which shouldn't rely on error messages.
How about adding an HtmlResponse property to the ShopifyException class?

**Comment by nozzlegear** [comment]
Yes, hopefully most people are using the status to check errors. But it does make it harder to debug problems when it's not showing the right error messages.




Adding HtmlResponse would be good, but first I'm trying to figure out if this change was intended or just some kind of Shopify bug.




--


Josh Harms







On Tue, Apr 17, 2018 at 6:35 PM -0500, "Clement Gutel" <notifications@github.com> wrote:












Hopefully it shouldn't break any applications, which shouldn't rely on error messages.

How about adding an HtmlResponse property to the ShopifyException class?

**Comment by clement911** [comment]
Oh ok thanks for that. I'm curious to know.

**Comment by nozzlegear** [comment]
Got another instance of this happening here: https://ci.appveyor.com/project/nozzlegear/shopifysharp/build/4.0.190

I've tried to contact Shopify's support about this, but last time it took them six weeks to get back to me. 🙃
